index.js

```diff
@@ -319,78 +265,65 @@ module.exports = {
 
-    var outputPath = '/' + cleanPath(this.outputDir) + '/' + this.outputFileName;
-    var paths = this.app.options.outputPaths;
+    removeFromTree = function(inputFiles) {
+      workingTree = fileRemover(workingTree, {
+        files: inputFiles
+      });
+    };
 
-    /* Locate all script files and concatenate into one file. */
+    /* Locate all script files and concatenate into one file */
 
-    var scriptInputFiles = [
-      cleanPath(paths.vendor['js'])
-    ];
+    if (jsOptions.concat) {
+      scriptInputPaths = this.filterAndCleanPaths('js');
 
-    if (testing) {
-      scriptInputFiles.push(cleanPath(paths.testSupport['js']));
+      concatenatedScripts = concatAndMap(tree, {
+        allowNone: true,
+        inputFiles: scriptInputPaths,
+        outputFile: outputPath + '.js',
+        footer: jsOptions.footer,
+        header: jsOptions.header,
+        wrapInFunction: this.wrapScriptsInFunction
+      });
     }
 
-    scriptInputFiles.push(cleanPath(paths.app['js']));
-
-    var concatenatedScripts = concat(tree, {
-      allowNone: true,
-      inputFiles: scriptInputFiles,
-      outputFile: outputPath + '.js',
-      footer: this.footer,
-      header: this.header,
-      wrapInFunction: this.wrapScriptsInFunction
-    });
-
     /* Locate all style files and concatenate into one file */
 
-    var styleInputFiles = [
-      cleanPath(paths.vendor['css'])
-    ];
+    if (cssOptions.concat) {
+      styleInputPaths = this.filterAndCleanPaths('css');
 
-    if (testing) {
-      styleInputFiles.push(cleanPath(paths.testSupport['css']));
+      concatenatedStyles = concatAndMap(tree, {
+        allowNone: true,
+        footer: cssOptions.footer,
+        header: cssOptions.header,
+        inputFiles: styleInputPaths,
+        outputFile: outputPath + '.css',
+        wrapInFunction: false
+      });
     }
 
-    var appCssPaths = paths.app['css'];
+    /* Combine all the files into the project's tree */
 
-    /* The app tree's CSS uses a KVP relationship so we have to do a little more work than we did for the scripts... */
+    trees = [tree, concatenatedScripts, concatenatedStyles];
 
-    for (var path in appCssPaths) {
-      styleInputFiles.push(cleanPath(appCssPaths[path]));
-    }
+    /* Remove empty values and add the remaining to the
+    working tree */
 
-    var concatenatedStyles = concat(tree, {
-      allowNone: true,
-      footer: this.footer,
-      header: this.header,
-      inputFiles: styleInputFiles,
-      outputFile: outputPath + '.css',
-      wrapInFunction: false
-    });
+    workingTree = mergeTrees(trees.filter(function(e) { return e; }));
 
-    /* Combine all the files into the project's tree */
+    /* Remove the unnecessary script and style files */
 
-    var workingTree = mergeTrees([tree, concatenatedScripts, concatenatedStyles]);
+    if (jsOptions.concatScripts && jsOptions.preserveOriginal) {
```

Comment:
same here - not sure if the logic for preserveOriginal is correct - still grokking the context